### PR TITLE
[[ Bug 14457 ]] Split TRANSLATED_PATH when additional segments follow th...

### PR DESCRIPTION
**Pull request by livecodesebastien**
...e filepath

### CODE REVIEW

**Comment by runrevmark** [comment]
@runrevsebastien: Can you give an example of TRANSLATED_PATH which was processed wrongly, and how it should be processed?

**Comment by livecodesebastien** [comment]
Yes: in CGI mode, a URL like `http://localhost/helloworld.lc/other/things` does not work with 7.0 (but does with 6.7).

**Comment by livecodesebastien** [comment]
Updated - fix an issue and more comments, hopefully explaining in a clearer way what we do to the TRANSLATED_PATH.

File: engine/src/srvcgi.cpp [hunk 1]
```diff
@@ -707,12 +707,11 @@ static void cgi_store_form_urlencoded(MCVariable *p_variable, MCDataRef p_data,
...
-	char *t_path, *t_path_end;
+    char *t_path;
 
 	MCStringRef env;
     env = nil;
-	t_path = nil;
-    t_path_end = nil;
+    t_path = nil;
     
     // SN-2014-07-29: [[ Bug 12865 ]] When a LiveCode CGI script has a .cgi extension and has
```
Comment (runrevmark) [inline engine/src/srvcgi.cpp:716]: I'm not clear here what you mean - the same processing seems to be being applied to t_path regardless of whether it came from PATH_TRANSLATED or SCRIPT_FILENAME.

Looking at this - http://stackoverflow.com/questions/2261951/what-exactly-is-path-info-in-php - it suggests that if PATH_INFO is not being set then its a problem with web-server config not the thing processing the CGI request.

Comment (livecodesebastien) [inline engine/src/srvcgi.cpp:716]: That's simply for a backwards-compatibility that this was reinstated, as this has always been done by the engine itself, not Apache (or such).
As Apache doc seems to say (http://httpd.apache.org/docs/2.2/mod/core.html#acceptpathinfo), I'm not sure that the PATH_INFO must be set - but more that extra segments are allowed. Even with the AcceptPathInfo explicitly set to 'on', the URL described in the bug report is not working with LC 7.0.1

Comment (runrevmark) [inline engine/src/srvcgi.cpp:716]: Okay cool - as long as things work the same as they did before then that's fine.

Perhaps file a bug for later consideration about removing the engine-specific behavior if it is indeed fixable by web-server configuration.

**Comment by runrevmark** [comment]
Reviewed-by: @runrevmark.

**Pull request merged**